src/readers/morphologyHDF5.cpp

```diff
@@ -216,32 +216,33 @@ void MorphologyHDF5::_readPoints(int firstSectionOffset) {
     auto& somaPoints = _properties._somaLevel._points;
     auto& somaDiameters = _properties._somaLevel._diameters;
 
-    auto loadPoints = [&](const std::vector<std::array<float, _pointColumns>>& hd5fData,
-                          bool hasNeurites) {
-        const std::size_t section_offset = hasNeurites ? std::size_t(firstSectionOffset)
-                                                       : hd5fData.size();
-
-        // points and diameters are PODs. Fastest to resize then assign values
-        somaPoints.resize(somaPoints.size() + section_offset);
-        somaDiameters.resize(somaDiameters.size() + section_offset);
-        for (std::size_t i = 0; i < section_offset; ++i) {
-            const auto& p = hd5fData[i];
-            somaPoints[i] = {p[0], p[1], p[2]};
-            somaDiameters[i] = p[3];
-        }
-
-        if (hasNeurites) {
-            const size_t size = (points.size() + hd5fData.size() - section_offset);
-            points.resize(size);
-            diameters.resize(size);
-            for (std::size_t i = section_offset; i < hd5fData.size(); ++i) {
+    auto loadPoints =
+        [&](const std::vector<std::array<morphio::floatType, _pointColumns>>& hd5fData,
+            bool hasNeurites) {
```

Comment:
```suggestion
    auto loadPoints = [&](
        const std::vector<std::array<morphio::floatType, _pointColumns>>& hd5fData,
        bool hasNeurites
    ) {
```
Embrace it… then keep indentation as before.

Reply:
Nope